CONTRIBUTING.md

```diff
@@ -8,144 +8,189 @@ The following is a short set of guidelines for contributing to Iroha.
+- Unless otherwise specified refer to [rust's best practices](https://github.com/mre/idiomatic-rust).
+- Use the `mod.rs` style. [Self-named modules](https://rust-lang.github.io/rust-clippy/master/) will not pass static analysis, except as [`trybuild`](https://crates.io/crates/trybuild) tests.
+- Use a domain-first modules structure. (Example: don't do `constants::logger`, instead invert the hierarchy, putting the object for which it is used first: `iroha_logger::constants`).
+- Use [`expect`](https://learning-rust.github.io/docs/e4.unwrap_and_expect.html) with an explicit error message or proof of infallibility instead of `unwrap`.
+- Use only full words in *public* structure, variable, method, trait, constant and module names.
+- However, abbreviations are allowed if:
+  - The name is local (e.g. closure arguments).
+  - The name is abbreviated by rust convention (e.g. `len`, `typ`).
+  - The name is an accepted abbreviation (e.g. `tx`, `wsv` etc) TODO link glossary.
+  - The full name would have been shadowed by a local variable (e.g. `msg <- message`).
+  - The full name would have made the code cumbersome (no more than 5-6 words in name) (e.g. `WorldStateViewReceiverTrait -> WSVRecvTrait`).
+  - If in doubt: clearer is better than shorter.
+- Never ignore an error. If you can't panic and can't recover, it at least needs to be recorded in the log.
+- Prefer to return a `Result` instead of `panic!`.
+  - Except when implementing something that uses `issue_send` instead of `send` ([more about actors](docs/source/guides/actor.md)). Actors and parallelism don't mix; you could deadlock the entire peer, so it's better to `panic!` if something goes wrong. This is a necessary concession for asynchronous programming.
+- Group related functionality spatially, preferably inside appropriate modules. For example, instead of having a block with `struct` definitions and then `impl`s for each individual struct, it is better to have the `impl`s related to that `struct` next to it.
+- Otherwise:  declaration before implementation;`use` statements at the top.
+- When writing non-doc comments, instead of describing *what* your function does, try to explain *why* it does something in a particular way. This will save you and the reviewer time.
+- Try to avoid `use` statements if the imported name is used only once. This makes moving your code into a different file easier.
+- Do not silence clippy lints indiscriminately. If you do, explain your reasoning with a comment (or `expect` message).
+- We use pinned dependencies. If your work depends on a particular crate, see if it wasn't already installed using [`cargo tree`](https://doc.rust-lang.org/cargo/commands/cargo-tree.html) (hint use `bat` or `grep`), and try to use that version, instead of the latest version.
+- Version bumps in separate PR.
```

Comment:
could you add that we prefer full format(`"x.y.z"`) for our dependency versions? as opposed to `"x.y"` or `"x"`. This is actually implicitly caret requirement like `"^x.y.z"` and will make `cargo update` bump both patch and minor versions

I think this alone is sufficient and we should only allow other version requirements in justified special cases